### DISCUSSION TITLE
I want to remove redundant method_exists() check

### DISCUSSION THREAD

**Discussion by hrant1020** [laravel/framework#51283] [Show and tell] [answered]
I want to remove the unnecessary code of `method_exists($enumClass, 'tryFrom')`. I belive the refactoring is correct. I'd like to know your opinion: could it potentially break something ? 

PR  link  https://github.com/laravel/framework/pull/51282/

enum() method  from \Illuminate\Http\Concerns\InteractsWithInput    

![image](https://github.com/laravel/framework/assets/28590762/27367940-28d7-48ee-b4a4-2769102c2f7a)

As you see when `$enumClass`  is not "enum type" method returns null  so I think we can remove redundant method check `method_exists($enumClass, 'tryFrom')`  because if it is enum it by default have that method if it is not enum then method returns null

So I have done  those changes PR  link  https://github.com/laravel/framework/pull/51282/

and this is method from PR

```PHP
    /**
     * Retrieve input from the request as an enum.
     *
     * @template TEnum
     *
     * @param  string  $key
     * @param  class-string<TEnum>  $enumClass
     * @return TEnum|null
     */
    public function enum($key, $enumClass)
    {
        if ($this->filled($key) && enum_exists($enumClass)) {
            return $enumClass::tryFrom($this->input($key));
        }

        return null;
    }
```

I think it is correct refactoring. I have also added a test for the enum() method's handling of empty value requests.

**Comment by mbabker** [comment, marked as answer]
The root enum class, [`UnitEnum`](https://www.php.net/manual/en/class.unitenum.php), does not have a `tryFrom()` method.  This is only part of the [`BackedEnum`](https://www.php.net/manual/en/class.backedenum.php) class.  So technically the `method_exists()` check, even if it could be expressed better but probably won't due to Laravel's liberal use of duck-typing in the framework, is required.  I guess technically this way one could create a non-backed enum and implement a custom `tryFrom()` method, but that seems like such an off edge case for PHP as a whole that there's little benefit to the framework allowing this.

Even the generic template on the method leaves the type narrowing out of its declaration, so passing any PHP class string there would be legal, so the usefulness of that is even minimal at best.

**Reply by henzeb** [reply]
It does allow quite a bit... In new software a case can be written redundant... "case Admin='Admin';" looks foolish, even when using lowercase. Maybe one day they will add tryFrom to unitEnums
